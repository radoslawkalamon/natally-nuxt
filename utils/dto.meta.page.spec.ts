import { DTOMetaPage } from '@/utils/dto.meta.page'

describe('Utils / DTO / Meta / Page', () => {
  describe('Page path with slash', () => {
    let dtoMetaPage: DTOMetaPage

    beforeAll(() => {
      dtoMetaPage = new DTOMetaPage({
        createdAt: '2021-11-01T00:00:00+00:00',
        description: 'Test Description',
        title: 'Test title',
        updatedAt: '2021-11-01T00:00:00+00:00',
        path: '/test-url/'
      })
    })

    test('shall match snapshot', () => {
      expect(dtoMetaPage).toMatchSnapshot()
    })
  })

  describe('Page path without slash', () => {
    let dtoMetaPage: DTOMetaPage

    beforeAll(() => {
      dtoMetaPage = new DTOMetaPage({
        createdAt: '2021-11-01T00:00:00+00:00',
        description: 'Test Description',
        title: 'Test title',
        updatedAt: '2021-11-01T00:00:00+00:00',
        path: '/test-url'
      })
    })

    test('shall match snapshot', () => {
      expect(dtoMetaPage).toMatchSnapshot()
    })
  })
})
