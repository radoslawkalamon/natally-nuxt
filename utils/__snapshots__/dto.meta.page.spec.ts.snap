// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils / DTO / Meta / Page Page path with slash shall match snapshot 1`] = `
DTOMetaPage {
  "createdAt": 2021-11-01T00:00:00.000Z,
  "description": "Test Description",
  "path": "/test-url/",
  "title": "Test title",
  "updatedAt": 2021-11-01T00:00:00.000Z,
}
`;

exports[`Utils / DTO / Meta / Page Page path without slash shall match snapshot 1`] = `
DTOMetaPage {
  "createdAt": 2021-11-01T00:00:00.000Z,
  "description": "Test Description",
  "path": "/test-url/",
  "title": "Test title",
  "updatedAt": 2021-11-01T00:00:00.000Z,
}
`;
